Tools/CopyPermalink/Sublime Text/CopyWebKitPermalink/CopyWebKitPermalink.py

```diff
@@ -80,19 +81,21 @@ def run(self, edit, annotate_blame=False):
             return
 
         document_path = self.view.file_name()
-        self._last_svn_info = None
         self._directory_in_checkout = document_path if os.path.isdir(document_path) else os.path.dirname(document_path)
-        self.determine_vcs_from_path(document_path)
 
-        if s_settings.include_revision and not self.path_is_in_webkit_checkout(document_path):
+        if not self.is_in_webkit_checkout():
             return
 
+        mode = "blame" if annotate_blame else "blob"
+
+        branch = self.current_git_commit() if s_settings.match_commit else self.current_git_branch()
+
+        path = self.path_relative_to_repository_root_for_path(document_path)
+
         line_number, _ = self.view.rowcol(self.view.sel()[0].begin())  # Zero-based
         line_number = line_number + 1
 
-        path = self.path_relative_to_repository_root_for_path(document_path)
-        revision_info = self.revision_info_for_path(document_path)
-        permalink = self.permalink_for_path(path, line_number, revision_info, annotate_blame)
+        permalink = 'https://github.com/WebKit/WebKit/{}/{}/{}#L{}'.format(mode, branch, urllib.parse.quote(path), line_number)
```

Comment:
Any reason we're restricting this to WebKit? It should generally work for any GitHub project, yes?